flumine/execution/transaction.py

```diff
@@ -146,6 +168,9 @@ def replace_order(
     def execute(self) -> int:
         packages = []
         if self._pending_place:
+            for order, market_version in self._pending_place:
```

Comment:
We can't delay like this as it will result in the controls that act on live/open/count orders/trades incorrect

Reply:
I see that `RunnerContext.place` updates these fields:
 - invested
 - datetime_last_placed
 - trades
 - live_trades

I've updated the transaction to remove failed trades from live_trades, but am unsure whether the other 3 fields need to be reverted as well.

If a transaction contains multiple orders on the same runner, working out which datetime_last_placed to set it to could be tricky.

Similarly for RunnerContext.trades, if a trade contains multiple orders, but only one of them fails, which element from RunnerContext.trades should we remove?